### PR TITLE
IBX-8104: Document how to configure elastic search with analyzers for different languages

### PR DESCRIPTION
**Pull request by vidarl**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-8104](https://issues.ibexa.co/browse/IBX-8104)
| Versions      | `master`, `4.6`, ~~`3.3`~~
| Edition       | Experience, Commerce

Document how to specify language specific analyzers in ElasticSearch

Preview: https://ez-systems-developer-documentation--2366.com.readthedocs.build/en/2366/search/search_engines/elastic_search/configure_elastic_search/#add-language-specific-analysers

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [x] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/search/search_engines/elastic_search/configure_elastic_search.md [hunk 1]
```diff
@@ -475,6 +475,180 @@ ibexa_elasticsearch:
...
+
+This pattern matches on English. See [Define a template](#define-a-template) on how to specify the pattern for your language.
+
+#### Create config for language specific analyser.
+
+[Elastic Search documentation](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/analysis-lang-analyzer.html)
+explains how to configure analyzer for each specific language.
+
+Adopopting the [English analyzer](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/analysis-lang-analyzer.html#english-analyzer)
+will look like this in the [[= product_name =]] configuration:
```
Comment (adriendupuis) [inline docs/search/search_engines/elastic_search/configure_elastic_search.md:511]:
```suggestion
looks like this in the [[= product_name =]] configuration:
```

File: docs/search/search_engines/elastic_search/configure_elastic_search.md [hunk 2]
```diff
@@ -475,6 +475,180 @@ ibexa_elasticsearch:
...
 
+### Language specific analyzis
+
+It is possible to configure language specific analysis like stemming so that searching for "cars" will give hit on content that contain the word "car".
+On a multilingual site, it is possible to have different analyzing configured for different languages, something which
+typically is needed as stemming rules are language specific.
+
+#### Make a copy of the default template
+
+The first step in order to enable language specific analysis is to make a new template for each language in `config/packages/ibexa_elasticsearch.yaml`.
```
Comment (adriendupuis) [inline docs/search/search_engines/elastic_search/configure_elastic_search.md:486]:
```suggestion
The first step to enable language specific analysis is to make a new template for each language in `config/packages/ibexa_elasticsearch.yaml`.
```

**Review by adamwojs — APPROVED** [review]

File: docs/search/search_engines/elastic_search/configure_elastic_search.md [hunk 3]
```diff
@@ -475,6 +475,180 @@ ibexa_elasticsearch:
...
+                                type: long
+                    -   ez_mlong:
+                            match: "*_ml"
+                            mapping:
+                                type: long
++                    -   ez_text_fulltext:
++                            match: "*_fulltext"
++                            mapping:
++                                type: text
++                                analyzer: english_analyzer
```
Comment (vidarl) [inline docs/search/search_engines/elastic_search/configure_elastic_search.md:610]: With [this PR](https://github.com/ezsystems/ezplatform-elastic-search-engine/pull/57) merged, the diff will look slightly different in next versions.

Do we adjust for that now, or do it after the next releases of 3.3 and 4.6 ?

File: docs/search/search_engines/elastic_search/configure_elastic_search.md [hunk 4]
```diff
@@ -475,6 +475,180 @@ ibexa_elasticsearch:
...
+
+It is possible to configure language specific analysis like stemming so that searching for "cars" gives hit on content that contains the word "car".
+On a multilingual site, it is possible to have different analyzing configured for different languages, something which
+typically is needed as stemming rules are language specific.
+
+#### Make a copy of the default template
+
+The first step to enable language specific analysis is to make a new template for each language in `config/packages/ibexa_elasticsearch.yaml`.
+This template should be based on the `default` template found in `vendor/ibexa/elasticsearch/src/bundle/Resources/config/default-config.yaml`.
+The name of the new template should reflect the language, ie. `english`, `norwegian` or `french`,
```
[the remark this answers is not in the document]
Comment (vidarl) [reply docs/search/search_engines/elastic_search/configure_elastic_search.md:488]: In such case, it should read:
```suggestion
The name of the new template should indicate the language it applies to, for example `eng_gb`, `nor_no` or `fre_fr`.
```

**Review by vidarl — COMMENTED** [review]
Sorry for the delay here. I have one correction to your suggestions.. Feel free to commit it if you agree

**Review by dabrt — APPROVED** [review]

File: docs/search/search_engines/elastic_search/configure_elastic_search.md [hunk 5]
```diff
@@ -475,6 +475,179 @@ ibexa_elasticsearch:
...
++                            stopwords: "_english_"
++                        english_keywords:
++                            type: keyword_marker
++                            keywords: []
++                        english_stemmer:
++                            type: stemmer
++                            language: light_english
++                        "english_possessive_stemmer": {
++                          "type":       "stemmer",
++                          "language":   "possessive_english"
```
Comment (adriendupuis) [inline docs/search/search_engines/elastic_search/configure_elastic_search.md:562]:
```suggestion
+                          "type": "stemmer",
+                          "language": "possessive_english"
```

**Review by adriendupuis — APPROVED** [review]
Thank you @vidarl for writing this documentation and discussing my changes.

File: docs/search/search_engines/elastic_search/configure_elastic_search.md (hunk 3 above)
Comment (adriendupuis) [reply docs/search/search_engines/elastic_search/configure_elastic_search.md]: I used https://github.com/ibexa/elasticsearch/blob/v4.6.7/src/bundle/Resources/config/default-config.yaml#L73-L76 for the template so we're OK for this PR. I'll take time to check while adapting to 3.3.

**Review by dabrt — APPROVED** [review]

File: docs/search/search_engines/elastic_search/configure_elastic_search.md [hunk 6]
```diff
@@ -475,6 +475,46 @@ ibexa_elasticsearch:
...
+
+```yaml hl_lines="3-5 15-23 35 41-52 94 99"
+[[= include_file('code_samples/search/elasticsearch/config/packages/elasticsearch-en.yaml') =]]
+```
+
+Then, you must bind this language template to your Elasticsearch connection.
+
 ## Bind templates with connections
 
 Once you have created the field mapping template(s), you must establish a relationship between the templates and a connection. You do this by adding the "index_templates" key to a connection definition.
```
Comment (adriendupuis) [inline docs/search/search_engines/elastic_search/configure_elastic_search.md:520]:
```suggestion
If you have to create an index template (for example, for specific data types, or linguistic analysis), you must link this template to an Elasticsearch connection. You do this by adding the `index_templates` key to a connection definition.
```

@dabrt What do you think about this rewriting? Any counterproposal?

Comment (adriendupuis) [reply docs/search/search_engines/elastic_search/configure_elastic_search.md:520]:
```suggestion
After you create an index template (for example, for specific data types or linguistic analysis), you must link it to an Elasticsearch connection by adding the index_templates key to the connection definition.
```

**Pull request merged**